### PR TITLE
Add more types to settings page

### PR DESCRIPTION
**Pull request by Jellyfrog**
This allow us to validate the values are valid

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
Potential problems with `is_executable()`:
* Needs a full path
*  Path needs to be executable for the user running the PHP web process

File: LibreNMS/Util/DynamicConfigItem.php [hunk 1]
```diff
@@ -86,6 +86,10 @@ public function checkValue($value)
             return (bool)preg_match('/^#?[0-9a-fA-F]{6}([0-9a-fA-F]{2})?$/', $value);
         } elseif (in_array($this->type, ['text', 'password'])) {
             return !is_array($value);
+        } elseif ($this->type === 'executable') {
+            return is_executable($value);
```
Comment (murrant) [inline LibreNMS/Util/DynamicConfigItem.php:90]: Should probably check is_file() too so it doesn't get set to directories.

File: misc/config_definitions.json [hunk 2]
```diff
@@ -1538,16 +1538,16 @@
             "type": "boolean"
         },
         "fping": {
-            "default": "fping",
+            "default": "/usr/bin/fping",
```
Comment (murrant) [inline misc/config_definitions.json:1541]: Leave the defaults to the unqualified paths.  Binaries are expanded to full paths on first run:
https://github.com/librenms/librenms/blob/master/LibreNMS/Config.php#L442

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-66-release-changelog-july-2020/12953/1

**Pull request merged**